osu.Game/Graphics/UserInterface/SearchTextBox.cs

```diff
@@ -18,6 +18,12 @@ public SearchTextBox()
             PlaceholderText = HomeStrings.SearchPlaceholder;
         }
 
+        protected override void OnFocus(FocusEvent e)
+        {
+            base.OnFocus(e);
+            SelectAll();
```

Comment:
```suggestion
            if (HoldFocus)
                SelectAll();
```

Selecting all on focus feels good to use on desktop, probably only because of `FocusedTextBox` behaviour that never allows the user to manually focus the text box. If the text box were to be manually focused, it would be strange to ignore the clicked position (as this PR currently does).

But on mobile (you can test on desktop by modifying `FocusedTextBox` not to take focus eagerly), it feels different to use. Tapping on an unfocused textbox (eg. to fix a typo or to refine your search query) will require two taps with this change – one to focus it, and one to move the caret to the desired place. Previously, tapping it would only move the caret.
Maybe the new behaviour would be desired for some use cases...

Reply:
I've applied this to not bother mobile users in settings.